client/my-sites/site-settings/publishing-tools/post-by-email.tsx

```diff
@@ -0,0 +1,160 @@
+import isJetpackModuleActive from 'calypso/state/selectors/is-jetpack-module-active';
+import isJetpackModuleUnavailableInDevelopmentMode from 'calypso/state/selectors/is-jetpack-module-unavailable-in-development-mode';
+import isJetpackSiteInDevelopmentMode from 'calypso/state/selectors/is-jetpack-site-in-development-mode';
+import isRegeneratingJetpackPostByEmail from 'calypso/state/selectors/is-regenerating-jetpack-post-by-email';
+import isSiteAutomatedTransfer from 'calypso/state/selectors/is-site-automated-transfer';
+import { isJetpackSite } from 'calypso/state/sites/selectors';
+import { IAppState } from 'calypso/state/types';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
+type PostByEmailSettingProps = {
+	isFormPending: boolean;
+	address?: string;
+};
+
+type PostByEmailSettingComponentProps = PostByEmailSettingProps & {
+	selectedSiteId: number | null;
+	regeneratePostByEmail: ( siteId: number ) => void;
+	activateModule: ( siteId: number, moduleSlug: string ) => void;
+	deactivateModule: ( siteId: number, moduleSlug: string ) => void;
+	active: boolean;
+	moduleUnavailable: boolean | null;
+	regeneratingPostByEmail: boolean;
+	siteIsAtomic: boolean;
+	siteIsJetpack: boolean | null;
+};
+
+const moduleSlug = 'post-by-email';
+
+const PostByEmailSettingComponent = ( {
+	selectedSiteId,
+	isFormPending,
+	regeneratePostByEmail,
+	activateModule,
+	deactivateModule,
+	active,
+	moduleUnavailable,
+	regeneratingPostByEmail,
+	siteIsAtomic,
+	siteIsJetpack,
+	address,
+}: PostByEmailSettingComponentProps ) => {
+	const translate = useTranslate();
+
+	const handleRegenerate = () => {
+		if ( ! selectedSiteId ) {
+			return;
+		}
+
+		regeneratePostByEmail( selectedSiteId );
+	};
+
+	const email = address && address !== 'regenerate' ? address : '';
+	const labelClassName =
+		moduleUnavailable || regeneratingPostByEmail || ! active ? 'is-disabled' : undefined;
+
+	return (
+		<>
+			<FormFieldset>
+				<SupportInfo
+					text={ translate(
+						'Allows you to publish new posts by sending an email to a special address.'
+					) }
+					link={
+						siteIsAtomic
+							? localizeUrl( 'https://wordpress.com/support/post-by-email/' )
+							: 'https://jetpack.com/support/post-by-email/'
+					}
+					privacyLink={ ! siteIsAtomic }
+				/>
+				{ siteIsJetpack ? (
+					<JetpackModuleToggle
+						siteId={ selectedSiteId }
+						moduleSlug="post-by-email"
+						label={ translate( 'Publish posts by sending an email' ) }
+						disabled={ isFormPending || moduleUnavailable }
+					/>
+				) : (
+					<ToggleControl
+						checked={ active }
+						disabled={ isFormPending }
+						label={ translate( 'Post by Email' ) }
+						onChange={ ( checked ) => {
+							if ( ! selectedSiteId ) {
+								return;
+							}
+
+							if ( checked ) {
+								activateModule( selectedSiteId, moduleSlug );
+							} else {
+								deactivateModule( selectedSiteId, moduleSlug );
+							}
```

Comment:
I know the back-end for this isn't available yet, but conceptually, the `useQuery` and `useMutation` implementations in #90489 are great inspirations. Until the back-end is available, we can use mock implementations of those hooks.